Pet types list

Jessica-Mach opened the pull request:
User Story: List of Pet Types
Manpreet and Jessica

- created the PetType model with a constructor and a findAll method
- created the api/v1/pets/types endpoint
- created the PetTypesList to fetch from above endpoint
- created the PetTypeTile with Links that will go to pages that will display animals of that type
- incorporated schema and Seeder from Serge and Jack to ensure functionality
- NOTE: Jessica added a try/catch block and pool.end in the Seeder because data was doubling - seeder has a warning about calling pool.end() more than once even though it's only included once. May need review.

Jessica-Mach committed:
Added Pet Types List user story

Jessica-Mach committed:
Added pool.end and try/catch to Seeder to stop data from doubling during seeding process.

jackeymason commented on client/components/App.js:
```diff
@@ -1,17 +1,26 @@
 
 const App = props => {
   useEffect(() => {
     $(document).foundation()
```
Are we importing the foundation framework on line 13? Should we import at the application level instead rather than trying to load it with a page/component load?

dovakeen118 replied:
That looks fine as is with the Foundation import from "foundation-sites", and line 13 is initializing Foundation

jackeymason replied:
This was inherited code from LA.

jackeymason commented on client/components/App.js:
```diff
@@ -1,17 +1,26 @@
+  return (
+    <BrowserRouter>
+      <Switch>
+        <Route exact path="/" component={Home} />
+        <Route exact path="/pets" component={PetTypesList} />
```
Should both of these just point to {PetTypeList} so we can skip the redirect?

jackeymason replied:
URL doesn't get rewritten unless done this way.

jackeymason commented on server/db/Seeder.js:
```diff
@@ -1,15 +1,32 @@
 class Seeder {
   static async seed() {
-    // your seeder code here
+    let loadedData = fs.readFileSync('data.json');
+    let contents = JSON.parse(loadedData);
```
Serge - Here is that simple JSON data from file import I was thinking of: const data = require('.filePath/filename.json');
console.log(data); Nothing to change here, I was just really struck by this style of JSON data import when it came up in one of our mentor sessions. It lets us slide on import fs from 'fs'.

jackeymason reviewed (commented):
Nice work. I had a few questions we might discuss for informational purposes.

dovakeen118 commented on client/components/Home.js:
```diff
@@ -0,0 +1,11 @@
+const Home = (props) => {
+    
+    return(
+       <Redirect to="/pets" />
```
I would keep this component if it will be used for something in the future, but if it is just redirecting we can handle this in other ways

dovakeen118 commented on client/components/PetTypeTile.js:
```diff
@@ -0,0 +1,25 @@
+    let descriptionBlock
+    if(description) {
+        descriptionBlock= <p>{description}</p>;
+    }
```
not a requirement but a suggestion to possibly display filler text if there is no description
```
Description: N/A
```

dovakeen118 commented on client/components/PetTypesList.js:
```diff
@@ -0,0 +1,46 @@
+            <h1>Exotic Pets Adoption Agency</h1>
+            <h3>Types of Pets Available</h3>
+            {tiles}
+        </div>
```
Some of these files look like they have different tab size, it could also just be how the code is coming across on GitHub. Please have every team member go to their VSCode settings for both User and Workspace to set `Tab Size` to `2`. You can also look at Fang's staff announcement from this morning!

dovakeen118 commented on server/db/Seeder.js:
```diff
@@ -1,15 +1,32 @@
 import pg from "pg"
 import path from "path"
-import LineReader from "line-reader"
+//import LineReader from "line-reader"
```
this commented line can be removed!

dovakeen118 commented on server/db/Seeder.js:
```diff
@@ -1,15 +1,32 @@
+    let petTypes = contents.data.petTypes;
+    try{
+      petTypes.forEach(async (petType) => {
+          let query = 'INSERT INTO pet_types (type, img_url, description) VALUES ($1, $2, $3) RETURNING id';
```
not required but a suggestion to consider - this query will return the `id` of each new record, and in this instance for inserting, the `id` isn't used for anything! Here, we can omit the RETURNING clause

dovakeen118 commented on server/db/Seeder.js:
```diff
@@ -1,15 +1,32 @@
   }
 }
 
+Seeder.seed();
```
Shouldn't need to invoke the seeder here - looking at the `db:seed` script, it will look to `server/bin/seed.js`, which will call on the seeder

dovakeen118 commented on server/routes/api/v1/petsRouter.js:
```diff
@@ -0,0 +1,17 @@
+
+const petsRouter = new express.Router();
+
+petsRouter.get("/types", async (req, res) => {
```
if this router is meant for petTypes, I would suggest renaming it `petTypesRouter`. That way the path can also just be `/api/v1/petTypes` rather than `/api/v1/pets/types`

dovakeen118 reviewed (commented):
Looking great! Some small notes to consider

Jessica-Mach committed:
Made changes based on feedback of PR

Jessica-Mach commented:
Updated based on feedback on PR
- Fixed tabbing size
- Added N/A when a description is not available
- Updated the name of the petTypesRouter and api endpoint
- Fixed redirect to /pets to be handled by express instead of ReactRouter
- Removed the comment in the Seeder file
